### PR TITLE
User can now configure and use external services

### PR DESCRIPTION
**Pull request by bpagon13**

### CODE REVIEW

File: vent/api/actions.py [hunk 1]
```diff
@@ -683,26 +683,42 @@ def get_configure(self,
...
-            # load all vent.template options into dict
-            template_dict = {}
-            for section in tools[tool]:
-                template_dict[section] = json.loads(tools[tool][section])
-            # display all those options as they would in vent.template
-            return_str = ""
+        template_dict = {}
+        return_str = ""
+        if main_cfg:
+            vent_cfg_path = os.path.join(os.path.expanduser('~'), '.vent',
```
Comment (cglewis) [inline vent/api/actions.py]: there's a helper variable for this now

File: vent/api/actions.py [hunk 2]
```diff
@@ -683,26 +683,42 @@ def get_configure(self,
...
+                vent_cfg = Template(vent_cfg_path)
+                for section in vent_cfg.sections()[1]:
+                    template_dict[section] = {}
+                    for vals in vent_cfg.section(section)[1]:
+                        template_dict[section][vals[0]] = vals[1]
+            else:
+                status = (False, "Couldn't get vent.cfg information")
+        else:
+            # all possible vent.template sections
+            options = ['info', 'service', 'settings', 'docker', 'gpu']
```
Comment (cglewis) [inline vent/api/actions.py]: if this is all possible `vent.template` sections, why not just grab `sections`?

File: vent/api/actions.py [hunk 3]
```diff
@@ -734,67 +749,81 @@ def save_configure(self,
...
-                                                               options)
-            if tools:
-                tool = tools.keys()[0]
-            else:
-                status = (False, "Couldn't save configuration")
-        if status[0]:
-            try:
-                # save in vent.template
-                with open(template_path, 'w') as f:
+            vent_cfg_path = os.path.join(os.path.expanduser('~'), '.vent',
```
Comment (cglewis) [inline vent/api/actions.py]: use the helper variable here as well.

File: vent/api/plugin_helpers.py [hunk 4]
```diff
@@ -305,10 +305,33 @@ def start_sections(self,
             tool_d[c_name]['labels']['vent.version'] = version
             tool_d[c_name]['labels']['vent.name'] = s[section]['name']
 
-            log_config = {'type': 'syslog',
-                          'config': {'syslog-address': 'tcp://0.0.0.0:514',
-                                     'syslog-facility': 'daemon',
-                                     'tag': 'plugin'}}
+            # check for log_config settings in external-services
+            externally_configured = False
+            vent_config = Template(template=join(self.path_dirs.meta_dir,
```
Comment (cglewis) [inline vent/api/plugin_helpers.py]: and here.

File: vent/api/actions.py (hunk 2 above)
Comment (bpagon13) [inline vent/api/actions.py]: By sections, I meant sections in vent.template, but for grabbing configuration information, I'm looking at the plugin_manifest, which stores those as options. I could loop through all the options for each section in plugin_manifest, but I'd have to eventually ask if those options are equal to one of the ones listed in the list, so I just used constrained_sections with those options. Sorry, I realize the wording is a little misleading and am changing that.

Comment (cglewis) [inline vent/api/actions.py]: ah, i see.  gotcha.

File: vent/api/actions.py (hunk 1 above)
Comment (cglewis) [inline vent/api/actions.py]: @bpagon13 looks like this one got missed in your update, I think it's the last one.

Comment (bpagon13) [inline vent/api/actions.py]: Okay changed it

**Comment by cglewis** [comment]
this works mostly as expected, a couple things that should be improved before merging though:
- `Services` should be updated to report the correct thing
- it looks like you're expecting the end user to know how to configure docker `log_config`, rather than syslog and have vent do the translation, as it should.  currently it's inconsistent with the rest of `external-services`
- when starting/stopping/etc. containers that have been set to `locally_active` should not be listed as available options in the menus
- when `vent.cfg` is changed through the `system commands` menu, any running containers should be restarted to take into effect the changes made to `vent.cfg`, much like when `vent.template` files are changed.

File: vent/helpers/meta.py [hunk 5]
```diff
@@ -344,63 +344,95 @@ def Tools(**kargs):
...
+                                ip_address = settings_dict[setting]
+                            if setting == 'port':
+                                port = settings_dict[setting]
+                            if setting == 'protocol':
+                                protocol = settings_dict[setting]
+                        p.append(protocol + '://' + ip_address + ':' + port)
+                    if p and name:
+                        services.append((name, p))
+                except Exception as e:
+                    with open('/Users/bpagon/Desktop/random.txt', 'a') as f:
```
Comment (cglewis) [inline vent/helpers/meta.py]: this shouldn't be here...

Comment (cglewis) [inline vent/helpers/meta.py]: this shouldn't be here...

Comment (bpagon13) [inline vent/helpers/meta.py]: Whoops makeshift logger there, fixing now

**Comment by cglewis** [comment]
this breaks `build cores tools` for me:
```
2017-08-02 12:12:19,984 - vent.menus.tools - INFO - rq_worker:master:HEAD
2017-08-02 12:12:19,985 - vent.api.actions - INFO - Starting: build
2017-08-02 12:12:19,988 - vent.api.actions - INFO - {'branch': 'master', 'self': <vent.api.actions.Action instance at 0x104d68998>, 'enabled': 'yes', 'repo': None, 'version': 'HEAD', 'groups': 'core', 'name': 'rq_worker'}
2017-08-02 12:12:19,993 - vent.api.actions - INFO - {'cyberreboot:vent:/vent/core/rq_worker:master:HEAD': {'path': '/Users/clewis/.vent/plugins/cyberreboot/vent/vent/core/rq_worker', 'image_name': 'cyberreboot/vent-rq-worker:master'}}
2017-08-02 12:12:19,994 - vent.api.actions - INFO - Building cyberreboot:vent:/vent/core/rq_worker:master:HEAD ...
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - Starting: builder
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - install path: /Users/clewis/.vent/plugins/cyberreboot/vent/vent/core/rq_worker
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - image name: cyberreboot/vent-rq-worker:master
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - build: True
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - branch: master
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - version: HEAD
2017-08-02 12:12:19,994 - vent.api.plugins - INFO - current working directory: /Users/clewis/.vent/.internals/plugins/cyberreboot/vent
2017-08-02 12:12:19,995 - vent.api.plugins - INFO - Trying to pull cyberreboot/vent-rq-worker:master
2017-08-02 12:12:23,009 - vent.api.plugins - INFO - Pulling rq_worker
master: Pulling from cyberreboot/vent-rq-worker
Digest: sha256:8db1c55e5da2f66887deb5d7e902331849fd1efc20d7b3bc36d1f2b37ac64276
Status: Image is up to date for cyberreboot/vent-rq-worker:master

2017-08-02 12:12:23,018 - vent.api.plugins - INFO - (True, 'Pulled cyberreboot/vent-rq-worker:master')
2017-08-02 12:12:23,033 - vent.api.actions - ERROR - build failed with error: 'NoneType' object has no attribute 'write_config'
2017-08-02 12:12:23,034 - vent.api.actions - INFO - Status of build: False
2017-08-02 12:12:23,034 - vent.api.actions - INFO - Finished: build
```

**Pull request merged**